authors/apps/profiles/views.py

```diff
@@ -67,3 +70,125 @@ class AuthorsAPIView(ListView):
+            raise Exception('You cannot follow your own profile.')
+        query_result = Follow.objects.filter(
+            follower_id=follower_id, followed_id=self.profile_id)
+        if len(query_result) != 0:
+            raise Exception('You are already following {}.'.format(name))
+
+    def get_profile_to_follow(self, followed):
+        profile = Profile.objects.get(id=followed)
+        serializer = ProfileSerializer(profile)
+        profile = serializer.data
+        profile['following'] = True
+        return profile
+
+    def delete(self, request, username):
+        """
+        User unfollows an Author
+        """
+        user_id = User.objects.get(username=request.user.username).id
+        try:
+            followed_id = User.objects.get(username=username).id
+            profile_id = Profile.objects.get(user_id=followed_id).id
+            follow = Follow.objects.filter(
+                follower_id=user_id, followed_id=profile_id)
+            if len(follow) == 0:
+                raise Exception(
+                    'You cannot unfollow a user you are not following.')
+            follow.delete()
+            return Response(
+                {'message': 'You have successfully unfollowed {}.'.format(
+                    username)},
+                status=status.HTTP_204_NO_CONTENT
+            )
+        except Exception as e:
+            if isinstance(e, User.DoesNotExist):
+                return Response(
+                    {'error': 'No user with name {} exists.'.format(username)},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+            return Response(
+                {'error': str(e)}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+    def get(self, request, username):
+        """
+        Get all Authors a user is following
+        """
+        user_id = FollowersView().get_user_id(username)
+        follows = Follow.objects.filter(follower_id=user_id)
+        serializer = FollowSerializer(follows, many=True)
+        following = list()
+        for follow in serializer.data:
+            user_id = Profile.objects.get(id=follow['followed']).user_id
+            username = User.objects.get(id=user_id).username
+            following.append(username)
+        if len(following) == 0:
+            response = {'message': '{} has no following'.format(username)}
+        response = {'following_count': len(following), 'following': following}
+        return Response(response, status=status.HTTP_200_OK)
+
+
+class FollowersView(APIView):
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, username):
+        """
+        Get all Users following an Author
+        """
+        user_id = self.get_user_id(username)
+        profile_id = Profile.objects.get(user_id=user_id).id
+        followers = Follow.objects.filter(followed_id=profile_id)
+        serializer = FollowSerializer(followers, many=True)
+        followers = list()
```

Comment:
Thanks for the good work @joelethan, What do you think about sending the whole user profile and time since follow, instead of just a username. Let's look at the frontend implementation, we are going to need to display the image and the bio, not just a username.